pyproject.toml

```diff
@@ -1,11 +1,56 @@
 [build-system]
-requires = [
-    "setuptools >= 37, < 61",
-    "wheel",
+requires = ["hatchling",]
+build-backend = "hatchling.build"
+
+[project]
+name = "pyccel"
+dynamic = ["version"]
+readme = "README.md"
+license = "MIT"
+requires-python = ">= 3.8"
+authors = [
+    { name = "Pyccel development team" },
+]
+keywords = [
+    "math",
+]
+dependencies = [
+    "filelock >= 3.4.0",
     "numpy >= 1.16",
     "sympy >= 1.2",
     "termcolor >= 1.0.0",
     "textx >= 2.2",
-    "filelock >= 3.4.0",
 ]
-build-backend = "setuptools.build_meta"
+
+[project.optional-dependencies]
+test = [
+    "astunparse >= 1.6.0",
+    "coverage >= 3.1",
+    "mpi4py >= 3.0.0",
+    "pytest >= 2.7",
+    "pytest-xdist >= 1.16",
+    "scipy >= 1.5.0",
+    "tblib >= 1.7.0",
+]
+
+[project.scripts]
+pyccel = "pyccel.commands.console:pyccel"
+pyccel-clean = "pyccel.commands.pyccel_clean:pyccel_clean_command"
+pyccel-init = "pyccel.commands.pyccel_init:pyccel_init_command"
+
+[project.urls]
+Homepage = "https://github.com/pyccel/pyccel"
+
+[tool.hatch.version]
+path = "pyccel/version.py"
+
+[tool.hatch.build.targets.sdist]
+
```

Comment:
Why this empty section? Should we exclude some directories here?

Reply:
It does indeed seem to be superfluous. Fixed [b0f3b5a](https://github.com/pyccel/pyccel/pull/1638/commits/b0f3b5a3a779e09300161c25b71fc2b5c2bdaaf6)